x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -542,4 +550,29 @@ static DeprecationIssue checkMonitoringExporterPassword(
         final String url = "https://www.elastic.co/guide/en/elasticsearch/reference/7.7/monitoring-settings.html#http-exporter-settings";
         return new DeprecationIssue(DeprecationIssue.Level.CRITICAL, message, url, details, null);
     }
+
+    static DeprecationIssue checkClusterRoutingAllocationIncludeRelocationsSetting(final Settings settings,
+                                                                                   final PluginsAndModules pluginsAndModules,
+                                                                                   final ClusterState clusterState) {
+        DeprecationIssue nodeDeprecationIssue =
+            getClusterRoutingAllocationIncludeRelocationsSettingDeprecationIssue(settings, DeprecationIssue.Level.CRITICAL);
+        if (nodeDeprecationIssue != null) {
+            return nodeDeprecationIssue;
+        }
+
+        // The setting is dynamic so it can be defined in the ClusterState settings
+        return getClusterRoutingAllocationIncludeRelocationsSettingDeprecationIssue(
+            clusterState.metadata().settings(),
+            DeprecationIssue.Level.WARNING
+        );
```

Comment:
I would prefer to have this part of the check added separately to `DeprecationChecks.CLUSTER_SETTINGS_CHECKS`, to avoid having a deprecation issue per node.